### PR TITLE
Ability to install PyMongo from PIP

### PR DESCRIPTION
**Pull request by duhast**
It turned out that `python-pymongo` package for Ubuntu 14.04 contains outdated `pymongo` module (version 2.6.3) which uses obsolete authentication schema incompatible with MongoDB v2.6.9. This way, when it's used to create MongoDB users, the further authentication fails without [upgrading auth schema](http://docs.mongodb.org/manual/reference/command/authSchemaUpgrade/#dbcmd.authSchemaUpgrade) manually on each DB server.

So this PR includes an ability to install latest PyMongo (2.8 at the moment) via [PIP](https://pip.pypa.io/en/latest/) which works fine with latest MongoDB v2.6.9.

### CODE REVIEW

**Comment by UnderGreen** [comment]
Thank you! Very usefull patch.

File: tasks/install.deb.yml [hunk 1]
```diff
@@ -34,6 +34,18 @@
...
-  apt: pkg={{item}}
-  with_items: mongodb_additional_packages
+- name: Install PyMongo package
+  apt: pkg=python-pymongo state=latest
+  when: not mongodb_pymongo_from_pip
+
+- name: Install PIP
+  apt: pkg={{ item }}
+  with_items:
+    - python-dev
```
Comment (UnderGreen) [inline tasks/install.deb.yml]: Here is really needed python-dev package?

Comment (duhast) [inline tasks/install.deb.yml]: Well, when it's missing, `pip install pymongo` gives the following wraning:
```
    /tmp/pip_build_root/pymongo/setup.py:217: UserWarning:
    ********************************************************************
    WARNING: The bson._cbson extension module could not
    be compiled. No C extensions are essential for PyMongo to run,
    although they do result in significant speed improvements.
    The output above this warning shows how the compilation failed.
    
    Please see the installation docs for solutions to build issues:
    
    http://api.mongodb.org/python/current/installation.html
    
    Here are some hints for popular operating systems:
    
    If you are seeing this message on Linux you probably need to
    install GCC and/or the Python development package for your
    version of Python.
    
    Debian and Ubuntu users should issue the following command:
    
        $ sudo apt-get install build-essential python-dev
    
    Users of Red Hat based distributions (RHEL, CentOS, Amazon Linux,
    Oracle Linux, Fedora, etc.) should issue the following command:
    
        $ sudo yum install gcc python-devel
    
    If you are seeing this message on Microsoft Windows please install
    PyMongo using the MS Windows installer for your version of Python,
    available on pypi here:
    
    http://pypi.python.org/pypi/pymongo/#downloads
    
    If you are seeing this message on OSX please read the documentation
    here:
    
    http://api.mongodb.org/python/current/installation.html#osx
    ********************************************************************
```

**Comment by duhast** [comment]
It also would be good if someone will adapt this change for yum-based distros.

**Pull request merged**